utils/tracing/go/tracing.go

```diff
@@ -1,104 +1,116 @@
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/exporters/trace/zipkin"
 	"go.opentelemetry.io/otel/propagation"
 	"go.opentelemetry.io/otel/sdk/resource"
-	sdktrace "go.opentelemetry.io/otel/sdk/trace"
-	semconv "go.opentelemetry.io/otel/semconv"
-	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/semconv"
 )
 
-func initTracer(tp *sdktrace.TracerProvider) func() {
+func initTracer(tp *trace.TracerProvider) func() {
 	otel.SetTracerProvider(tp)
 	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
 	return func() {
-		_ = tp.Shutdown(context.Background())
+		err := tp.Shutdown(context.Background())
+		if err != nil {
+			log.Fatalf("Tracer shutdown error: %v", err)
+		}
 	}
 }
 
-func newZipkinExporter(url string, logger *log.Logger) *zipkin.Exporter {
+func newZipkinExporter(url string, logger *default_log.Logger) (*zipkin.Exporter, error) {
 	exporter, err := zipkin.NewRawExporter(
 		url,
 		zipkin.WithLogger(logger),
-		zipkin.WithSDKOptions(sdktrace.WithSampler(sdktrace.AlwaysSample())),
+		zipkin.WithSDKOptions(trace.WithSampler(trace.AlwaysSample())),
 	)
 	if err != nil {
-		log.Fatal(err)
+		log.Printf("warning: zipkin exporter experienced an error: %v", err)
 	}
-	return exporter
+	return exporter, err
 }
 
-// InitBasicTracer initialises a basic opentelemetry tracer using a zipkin exporter. The
-// exporter exports to the provided url and the tracer takes on the given name concatenated
-// with the host name (e.g. Knative pod name).
-func InitBasicTracer(url string, serviceName string) func() {
-	var logger = log.New(os.Stderr, "tracer-log", log.Ldate|log.Ltime|log.Llongfile)
+// InitBasicTracer initialises a basic OpenTelemetry tracer using a zipkin exporter. The
+// exporter sends span and trace information to the provided URL. The tracer uses the name
+// provided, concatenated with the name of the host (e.g., Knative pod name or docker container),
+// and all nodes / functions which use this tracer will be labeled with "-FU" to denote that they
+// are functions, which distinguishes them from knative queue proxies if used in a cluster.
+func InitBasicTracer(url string, serviceName string) (func(), error) {
+	log.SetFormatter(&log.TextFormatter{
+		TimestampFormat: ctrdlog.RFC3339NanoFixed,
+		FullTimestamp:   true,
+	})
+	var logger = default_log.New(os.Stderr, "tracer-log", default_log.Ldate|default_log.Ltime|default_log.Llongfile)
 	hostname, ok := os.LookupEnv("HOSTNAME")
 	if !ok {
 		hostname = ""
 	}
-	exporter := newZipkinExporter(url, logger)
-	tp := sdktrace.NewTracerProvider(
-		sdktrace.WithSampler(sdktrace.AlwaysSample()),
-		sdktrace.WithSyncer(exporter),
-		sdktrace.WithResource(resource.NewWithAttributes(
-			semconv.ServiceNameKey.String(fmt.Sprintf("%v@%v", serviceName, hostname)),
+	exporter, err := newZipkinExporter(url, logger)
+	tp := trace.NewTracerProvider(
```

Comment:
you cannot let it proceed here if there was an error! return the error to the caller